docs/diagrams/AddInterestSequenceDiagram.puml

```diff
@@ -0,0 +1,66 @@
+@startuml
+!include style.puml
+
+skinparam SequenceReferenceBackgroundColor White
+box Logic LOGIC_COLOR_T1
+participant ":Logic\nManager" as LogicManager LOGIC_COLOR
+participant ":AddressBook\nParser" as AddressBookParser LOGIC_COLOR
+participant ":AddInterest\nCommandParser" as AddInterestCommandParser LOGIC_COLOR
+participant ":AddInterest\nCommand" as AddInterestCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+participant "interestList:\nSet<Interest>" as Interest MODEL_COLOR
+
+end box
+
+[-> LogicManager : execute\n("addInt 1 anime")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand\n("addInt 1 anime")
+activate AddressBookParser
+
+create AddInterestCommandParser
+AddressBookParser -> AddInterestCommandParser
+activate AddInterestCommandParser
+
+
+create Interest
+AddInterestCommandParser -> Interest
+activate Interest
+Interest --> AddInterestCommandParser
+deactivate Interest
+AddInterestCommandParser --> AddressBookParser
+deactivate AddInterestCommandParser
+
+AddressBookParser -> AddInterestCommandParser : parse\n("1 anime")
+activate AddInterestCommandParser
+
+create AddInterestCommand
+AddInterestCommandParser -> AddInterestCommand: new\n(1, interestList)
+activate AddInterestCommand
+
+AddInterestCommand --> AddInterestCommandParser
+deactivate AddInterestCommand
+
+AddInterestCommandParser --> AddressBookParser
+deactivate AddInterestCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+AddInterestCommandParser -[hidden]-> AddressBookParser
+destroy AddInterestCommandParser
+
+AddressBookParser --> LogicManager
+deactivate AddressBookParser
+
+LogicManager -> AddInterestCommand : execute()
+activate AddInterestCommand
+
+ref over AddInterestCommand, Model : Add Interest
```

Comment:
Not sure if `sd Add Interest` diagram is required for `ref Add Interest`. 💭 
Refer to: https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/uml.html#reference-frames "Reference Frames".

Reply:
Yes added sd